### PR TITLE
[DO_NOT_MERGE] Help message

### PR DESCRIPTION
**Pull request by noodlemoodle**
add help message to property. 
usage
```
    {
      section: 'signingOfficerQuestionSection',
      name: 'isSigningOfficer',
      class: 'Boolean',
      helpMessage: 'IS_SIGNING_OFFICER_HELP', // name of message
    },
```

### CODE REVIEW

File: src/foam/u2/detail/SectionedDetailPropertyView.js [hunk 1]
```diff
@@ -231,6 +231,7 @@ foam.CLASS({
           var errorSlot = prop.validateObj && prop.validationTextVisible ?
             this.data.slot(prop.validateObj) :
             foam.core.ConstantSlot.create({ value: null });
+          // debugger;
```
Comment (noodlemoodle) [inline src/foam/u2/detail/SectionedDetailPropertyView.js]:
```suggestion
```

**Comment by Nauna** [comment]
SO Im thinking eventually after this merge - we systematically remove 'help' from sections and use 'helpMessages' solely.

Other comment - this looks good to me - although I'd wonder if 'helpMessage' shouldn't have an adapt or something to help and avoid the UI update in SectionedDetailPropertyView

**Comment by jlhughes** [comment]
Is it help or helpMessage which is defined in foam for the FObject or FObjectProperty?

**Comment by noodlemoodle** [comment]
I'm not sure what you mean.. help message was added in this pr, and help existed on property but not on fojbect @jlhughes

File: src/foam/u2/detail/SectionedDetailPropertyView.js [hunk 2]
```diff
@@ -249,7 +249,8 @@ foam.CLASS({
                     this.enableClass(self.myClass('error'), errorSlot);
                   })
                 .end()
-                .callIf(prop.help, function() {
+                .callIf(prop.help || ( prop.helpMessage && prop.sourceCls_[prop.helpMessage] ), function() {
+                  var help = prop.helpMessage && prop.sourceCls_[prop.helpMessage] ? prop.sourceCls_[prop.helpMessage] : prop;
```
Comment (kgrgreer) [inline src/foam/u2/detail/SectionedDetailPropertyView.js]: Why return 'prop' if the expression is false?

Comment (noodlemoodle) [inline src/foam/u2/detail/SectionedDetailPropertyView.js]:
```suggestion
                  var help = prop.helpMessage && prop.sourceCls_[prop.helpMessage] ? prop.sourceCls_[prop.helpMessage] : prop.help;
```

Comment (noodlemoodle) [reply src/foam/u2/detail/SectionedDetailPropertyView.js]: sorry, should be prop.help. updating @kgrgreer

**Comment by jlhughes** [comment]
> I'm not sure what you mean.. help message was added in this pr, and help existed on property but not on FObject @jlhughes

'help' is defined on foam.core.Property.  Can we use this property property please or is there a reason for new property?

**Comment by noodlemoodle** [comment]
> > I'm not sure what you mean.. help message was added in this pr, and help existed on property but not on FObject @jlhughes
> 
> 'help' is defined on foam.core.Property. Can we use this property property please or is there a reason for new property?

yes, but as far as I know 'help' will only take a String value as argument, whereas this will take the name of a message defined on the source class of the property, which we want for translation purposes right?

**Comment by jlhughes** [comment]
> > > I'm not sure what you mean.. help message was added in this pr, and help existed on property but not on FObject @jlhughes
> > 
> > 
> > 'help' is defined on foam.core.Property. Can we use this property property please or is there a reason for new property?
> 
> yes, but as far as I know 'help' will only take a String value as argument, whereas this will take the name of a message defined on the source class of the property, which we want for translation purposes right?

@hchoura @kgrgreer are we planning on support property properties such as 'help' in our locale work - now or future?

**Comment by kgrgreer** [comment]
Do not merge.

**Comment by kgrgreer** [comment]
Let's not do this way.

**Pull request closed without merging**